asserts/tool/sign_test.go

```diff
@@ -0,0 +1,190 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package tool_test
+
+import (
+	"bytes"
+	"testing"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/asserts/assertstest"
+	"github.com/snapcore/snapd/asserts/tool"
+)
+
+func TestTool(t *testing.T) { TestingT(t) }
+
+type signSuite struct {
+	keypairMgr asserts.KeypairManager
+	testKeyID  string
+}
+
+var _ = Suite(&signSuite{})
+
+func (s *signSuite) SetUpSuite(c *C) {
+	testKey, _ := assertstest.GenerateKey(752)
+
+	s.keypairMgr = asserts.NewMemoryKeypairManager()
+	s.keypairMgr.Put(testKey)
+	s.testKeyID = testKey.PublicKey().ID()
+}
+
+const (
+	modelYaml = `type: model
+authority-id: user-id1
+series: "16"
+brand-id: user-id1
+model: baz-3000
+architecture: amd64
+gadget: brand-gadget
+kernel: baz-linux
+store: brand-store
+required-snaps: [foo, bar]
+timestamp: 2015-11-25T20:00:00Z
+`
+)
+
+func expectedModelHeaders(a asserts.Assertion) map[string]interface{} {
+	return map[string]interface{}{
+		"type":              "model",
+		"authority-id":      "user-id1",
+		"series":            "16",
+		"brand-id":          "user-id1",
+		"model":             "baz-3000",
+		"architecture":      "amd64",
+		"gadget":            "brand-gadget",
+		"kernel":            "baz-linux",
+		"store":             "brand-store",
+		"required-snaps":    []interface{}{"foo", "bar"},
```

Comment:
because that's what the parsing produces (because it could also be a list of lists), and the two are not comparable for free